devicetypes/smartthings/zwave-thermostat.src/zwave-thermostat.groovy

```diff
@@ -507,7 +507,8 @@ def ping() {
 def switchMode() {
 	def currentMode = device.currentValue("thermostatMode")
 	def supportedModes = state.supportedModes
-	if (supportedModes) {
+	// Old version of supportedModes was as string, make sure it gets updated
+	if (supportedModes && supportedModes[0].size() > 1) {
```

Comment:
Same comment as above about whether or not it's possible for this list to be an empty list (which means it would pass if(supportedModes) but supportedModes[0] could still throw an NPE?